sources/driverdog/src/main.rs

```diff
@@ -76,9 +81,17 @@ struct LinkModulesArgs {}
 #[argh(subcommand, name = "load-modules")]
 struct LoadModulesArgs {}
 
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+/// Enum to hold the two types of configurations supported
+enum DriverType {
+    Linking(LinkingDriverConfig),
+    Copying(CopyingDriverConfig),
+}
```

Comment:
You have to be a bit careful with `untagged` enum parsing, since Serde just tries to parse the enums in order.

Can you add a test where you try to parse a `LinkingDriverConfig` that doesn't have any `objects_source` or `objects_files`? My guess is that it will load as a `CopyingDriverConfig`.

(Is that even a possible configuration in this domain?)

Reply:
Hmm, after reading the docstring again, I don't think that's a valid `LinkingDriverConfig`.